docs/introduction-bdd.md

```diff
@@ -0,0 +1,97 @@
+  
+  The BDD process looks like this:
+
+  A Subject-Matter-Expert (typically a business user) works with a Business Analyst to identify a business requirement. 
+  
+  *This is expressed as a story using the following template:*
+
+|As a Role
+|-
+|I request a Feature
+|To gain a Benefit
+|The speaker, who holds the Role, is the person who will gain the Benefit from the requested Feature.
+
+This can also be paraphrased variously as ...
+
+I want to achieve a specific Goal, and as a Role I should be able to accomplish this by performing Functionality.
+A Role invokes Feature to cause a Benefit[1]
+
+Software teams work best when the developers and business stakeholders
+are communicating clearly with one another. A great way to do that is to
+collaboratively specify the work that’s about to be done using automated
+acceptance tests.
+
+When the acceptance tests are written as examples, they stimulate people’s
+imaginations and help them see other scenarios they hadn’t previously
+considered.
+
+When the team write their acceptance tests collaboratively, they can develop
+their own ubiquitous language for talking about their problem domain. This
+helps them avoid misunderstandings.[2]
+   
+   
+   **"Behavior-driven development is about implementing an application by describing its behavior from the perspective of its stakeholders”** -- DAN North
+
+
+# BDD with Cucumber:
+     
+   Cucumber is a command-line tool. When you run it, it reads in your specifications from plain-language text files called features, examines them for scenarios to test, and runs the scenarios against your system. Each scenario is a list of steps for Cucumber to work through. So that Cucumber can understand these feature files, they must follow some basic syntax rules. The name for this set of rules is Gherkin.
+     
+ ![Cucumber_role](images/cucumber_core.png)
+   
+Along with the features, you give Cucumber a set of step definitions, which
+map the business-readable language of each step into Ruby code to carry out
+whatever action is being described by the step. In a mature test suite, the
+step definition itself will probably just be one or two lines of Ruby that delegate to a library of support code, specific to the domain of your application, that knows how to carry out common tasks on the system. Normally that will involve using an automation library, like the browser automation library Capybara, to interact with the system itself.[2]
+    
+
+### Example
+   What makes Cucumber to stand out from the crowd of other testing tools is that it has been designed specifically to ensure the acceptance tests can easily be read and written by anyone on the team. This reveals the true value of acceptance tests: as a communication and collaboration tool. The easy readability of Cucumber tests draws business stakeholders into the process, helping you really explore and understand their requirements.
```

Comment:
We want to kill the perception of Cucumber as a "testing tool", so let's avoid this term. Cucumber is a design/collaboration/development tool.